src/test/java/teammates/test/cases/search/InstructorSearchTest.java

```diff
@@ -0,0 +1,81 @@
+ * {@link InstructorSearchQuery}.
+ */
+public class InstructorSearchTest extends BaseSearchTest {
+    @Test
+    public void allTests() {
+        InstructorsDb instructorsDb = new InstructorsDb();
+
+        InstructorAttributes ins1InCourse1 = dataBundle.instructors.get("instructor1OfCourse1");
+        InstructorAttributes ins2InCourse1 = dataBundle.instructors.get("instructor2OfCourse1");
+        InstructorAttributes ins1InCourse2 = dataBundle.instructors.get("instructor1OfCourse2");
+        InstructorAttributes ins2InCourse2 = dataBundle.instructors.get("instructor2OfCourse2");
+        InstructorAttributes insInArchivedCourse = dataBundle.instructors.get("instructorOfArchivedCourse");
+        InstructorAttributes insInUnregCourse = dataBundle.instructors.get("instructor5");
+
+        ______TS("success: search for instructors in whole system; query string does not match anyone");
+
+        InstructorSearchResultBundle results =
+                instructorsDb.searchInstructorsInWholeSystem("non-existent");
+        verifySearchResults(results);
+
+        ______TS("success: search for instructors in whole system; query string matches some instructors");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("instructor1");
+        verifySearchResults(results, ins1InCourse1, ins1InCourse2);
+
+        ______TS("success: search for instructors in whole system; query string should be case-insensitive");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("InStRuCtOr2");
+        verifySearchResults(results, ins2InCourse1, ins2InCourse2);
+
+        ______TS("success: search for instructors in whole system; instructors in archived courses should be included");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("archived");
+        verifySearchResults(results, insInArchivedCourse);
+
+        ______TS("success: search for instructors in whole system; instructors in unregistered course should be included");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("instructor5");
+        verifySearchResults(results, insInUnregCourse);
+
+        ______TS("success: search for instructors in whole system; deleted instructors no longer searchable");
+
+        instructorsDb.deleteInstructor(ins1InCourse1.courseId, ins1InCourse1.email);
+        results = instructorsDb.searchInstructorsInWholeSystem("instructor1");
+        verifySearchResults(results, ins1InCourse2);
+    }
+
+    // verifies that search results match with expected output
+    // To ignore keys in comparisons, this method modifies instructors passed into it to set keys to null
+    private static void verifySearchResults(InstructorSearchResultBundle actual,
+            InstructorAttributes... expected) {
+        assertEquals(expected.length, actual.numberOfResults);
+        assertEquals(expected.length, actual.instructorList.size());
+        standardizeInstructorsForComparison(expected);
```

Comment:
Changing the parameter `expected` itself is not a good practice. But it depends as this will only be used in the test case, let's look what others say.